### PR TITLE
add completion supports to istioctl experimental describe pod/service

### PR DESCRIPTION
**Pull request by silenceshell**
Add completion supports to istioctl experimental.

```bash
$ istioctl x describe service <tab><tab>
go-proxy   go-server  nginx
$ istioctl x describe pod <tab><tab>
debian-77dc9c5f4f-vrwj5                httpbin-66cdbdb6c5-l58q7               nginx-74f5bd857c-9hh2c
```

FYI: This PR used to try add completion support to istioctl "-i ns" and "-n ns", but the `RegisterFlagCompletionFunc` in cobra package will write to map `flagCompletionFunctions`, and this map is global and with no write lock, thus will case a lot of `concurrent map writes` errors in many test cases, so I will try to fix this bug of cobra first.

---

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by ericvn** [comment]
I believe we do need to add some release note text for this change.

**Comment by silenceshell** [comment]
> I believe we do need to add some release note text for this change.

Thanks, I have removed the `release-note-none` label.

**Comment by silenceshell** [comment]
@shamsher31 Would you please to review this PR? This is similar to #33367 .

File: istioctl/cmd/completion.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	for _, pod := range podList.Items {
+		if toComplete == "" || strings.HasPrefix(pod.Name, toComplete) {
+			podsName = append(podsName, pod.Name)
+		}
+	}
+
+	return podsName, nil
+}
+
+func validPodsNameArgsFunction(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```
Comment (shamsher31) [inline istioctl/cmd/completion.go:50]: nit:
```suggestion
func validPodsNameArgs(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

File: istioctl/cmd/completion.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	for _, service := range serviceList.Items {
+		if toComplete == "" || strings.HasPrefix(service.Name, toComplete) {
+			serviceNameList = append(serviceNameList, service.Name)
+		}
+	}
+
+	return serviceNameList, nil
+}
+
+func validServiceArgsFunction(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```
Comment (shamsher31) [inline istioctl/cmd/completion.go:85]: nit:
```suggestion
func validServiceArgs(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

**Review by shamsher31 — COMMENTED** [review]
Overall LGTM.

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**